### PR TITLE
Fix memory leaks in minigzip.

### PR DESCRIPTION
**Pull request by mtl1979**
See #786

### CODE REVIEW

File: test/minigzip.c [hunk 1]
```diff
@@ -162,10 +162,14 @@ void gz_uncompress(gzFile in, FILE *out) {
 
     for (;;) {
         len = PREFIX(gzread)(in, buf, BUFLENW);
-        if (len < 0) error (PREFIX(gzerror)(in, &err));
+        if (len < 0) {
+            free(buf);
+            error (PREFIX(gzerror)(in, &err));
```
Comment (nmoinvaz) [inline test/minigzip.c]: Remove space after error?

**Comment by nmoinvaz** [comment]
I think these same changes will be needed in minigzip_fuzzer as a duplicate function exists there.

**Comment by mtl1979** [comment]
@nmoinvaz The version in minigzip_fuzzer is not changed yet to use malloc... it still allocates the buffer on stack...

File: test/minigzip.c (hunk 1 above)
Comment (mtl1979) [reply test/minigzip.c]: Can do style fixes ;)

**Review by nmoinvaz — APPROVED** [review]

**Pull request merged**